src/Servers/IIS/IIS/src/Core/IISHttpContext.cs

```diff
@@ -118,11 +118,17 @@ internal unsafe IISHttpContext(
     public SslProtocols Protocol { get; private set; }
     public TlsCipherSuite? NegotiatedCipherSuite { get; private set; }
     public string SniHostName { get; private set; } = default!;
+    [Obsolete("Obsolete on SslStream.")]
```

Comment:
Do we want to use the same DiagId on these obsoletions as the ones used on SslStream? That way we (and users) don't need to suppress both `618` and `SYSLIB0058`.

Reply:
We don't use DiagId with obsolete attributes in ASP.NET Core: https://github.com/search?q=repo%3Adotnet%2Faspnetcore+%22%5BObsolete%22&type=code&p=1

I don't think so.

Reply:
Has there ever been a case where we need to Obsolete our API because of an underlying Obsoletion in the runtime libraries?

It feels like this is just an extension of obsoleting the underlying SslStream enums, so it feels like we should use the same DiagId.

cc'ing some devs who might have opinions - @terrajobst @bartonjs @rzikm @wfurt @halter73

Reply:
I agree with @eerhardt. There are two options to deal with obsoletions in underlying APIs:

1. **Suppression**. If the obsoletion is in the implementation and we intend to shield the consumer by eventually moving off of, then suppression on our end is fine.
2. **Mark as obsolete**. If the obsoletion is part of the API or conceptually tied to its semantics, then we should obsolete the higher level API. 

For (2) it makes little sense to me to use a different diagnostic ID, because they are logically due to the same obsoletion. If a customer suppresses one, I think they have reasonable expectation to suppress the other.

And generally speaking I believe we should virtually never obsolete without a diagnostic ID. The standard diagnostic for obsoletion applies way to broadly so we generally want to avoid cases where customers feel like should suppress all obsoletions.

Reply:
Ok, I added the diagnostic ID.

Reply:
Thanks, appreciate it!